README.md

```diff
@@ -1,104 +1,310 @@
 
-Decision DSL excerpt
-![DSL excerpt](./docs/readme/screenshot-dsl.png)
+# Architecture of the FAW
+
+Exploring data formats via input corpora at scale requires an
+infrastructure to run parsers and associated tools against input files
+in the corpus and capture their behavior. The FAW is divided in to a
+backend that handles the details of running parsers against the input
+corpus, and a frontend that enables users to investigate the behavior of
+parsers and tools against files in the corpus, at different scales and
+granularities.
+
+Parsers and parser-like tools are provided as binary executables and can
+be written in any language, and using any technology. Further FAW
+supports processing the file corpus either on a single machine or in
+parallel on a cluster of machines. The latter, termed a "teaming" setup,
+can be performed on a private network without requiring internet access.
+
+The FAW allows the user to specify analysis sets to control the scale
+and speed of analyses. It also provides multiple tools are provided to
+explore the interactions between inputs and parsing programs in the
+analysis set.
+
+![FAW Architecture](./docs/readme/faw-arch.png)
+
+# Quick Start
+
+Running the FAW requires an installation of Python (\>= 3.8). In order
+to start an instance:
+
+- Clone the repository
+- Install the prerequisites with `pip3 install -r requirements.txt`
+- Launch the startup script `workbench.py` with two paths: the path to a
+  *distribution folder* and one to the file corpus being studied.
+
+A FAW distribution is a specific configuration of the FAW and is usually
+associated with a specific file format. The FAW repository
+includes a small set of distributions and corresponding file corpora,
+for example PDF. The PDF distribution can be launched from the root
+folder with the following command: `workbench.py pdf test_files/pdf`.
+
+The workbench script builds a docker image for the distribution on the
+fly and launches an instance of the FAW. The FAW user interface is
+accessible via a webbrowser at `http://localhost:8123` (by default).
+
+By default, FAW is launched in [developer mode](#developer-friendly-features).
+
+
+In addition to directly running the distribution, the workbench script is capable of building and saving a unified docker image for a FAW distribution, which can can simplify deployment. Additionally, the FAW supports deployment in a multi-server "teaming mode" using [pyinfra](https://pyinfra.com/), which is useful when working with large corpora. The repository includes the `workbench-teaming-init.py` script for creating the pyinfra inventory/deploy files; pyinfra tooling can then be used to install prerequisites on target machines and set up FAW images and services. In this mode, target machines process files independently, but share a single database. The FAW UI is made available via a webserver on the database's machine.
 
-## Getting Started
+# FAW Concepts
```

Comment:
One concept that should be added here: Analysis sets. Also, maybe creating an analysis set should be one of the steps under "Getting Started" (or should be detailed here and linked to from there). IMO we don't need a full tutorial, just a definition.

For color: I'm also planning to tweak the frontend a bit to make creating an analysis set when there are none more obvious, since it's been a source of confusion to new users. This change probably won't be until after this PR is done though.

Reply:
Looks great!